packages/stencil/components/pagination/src/components/osds-pagination/osds-pagination.e2e.ts

```diff
@@ -0,0 +1,193 @@
+  }
+
+  describe('defaults', () => {
+    beforeEach(async () => {
+      await setup({ attributes: { totalPages: 10, current: 4 } });
+    });
+
+    it('should render', async () => {
+      expect(el).not.toBeNull();
+      expect(el).toHaveClass('hydrated');
+    });
+    it('should have arrows', async () => {
+      expect(liPaginationPageButtonElement).toHaveClass('arrows');
+    });
+    it('should have page list', async () => {
+      expect(liPaginationPageButtonElement).not.toBeNull();
+    });
+    it('should have page list with osds-buttons', async () => {
+      osdsButtonPaginationPageButtonElement = await page.find('osds-pagination >>> div > ul > li > osds-button');
+      expect(osdsButtonPaginationPageButtonElement).not.toBeNull();
+    });
+    it('arrows should have osds-icons', async () => {
+      let icons = await page.find('osds-pagination >>> div > ul > li > osds-button >>> osds-icons');
+      expect(icons).not.toBeNull();
+    });
+  });
+
+  describe('check the pagination structure', () => {
+    it('< 1 > should have 3 li', async () => {
+      await setup({ attributes: { current: 1, totalPages: 1 } });
+
+      const linkList = await page.findAll('osds-pagination >>> li');
+      expect(linkList.length).toBe(3);
+    });
+
+    it('< 1 2 > should have 4 li', async () => {
+      await setup({ attributes: { current: 1, totalPages: 2 } });
+
+      const linkList = await page.findAll('osds-pagination >>> li');
+      expect(linkList.length).toBe(4);
+    });
+
+    it('< 1 2 3 > should have 5 li', async () => {
+      await setup({ attributes: { current: 1, totalPages: 3 } });
+
+      const linkList = await page.findAll('osds-pagination >>> li');
+      expect(linkList.length).toBe(5);
+    });
+
+    it('< 1 2 3 4 > should have 6 li', async () => {
+      await setup({ attributes: { current: 1, totalPages: 4 } });
+
+      const linkList = await page.findAll('osds-pagination >>> li');
+      expect(linkList.length).toBe(6);
+    });
+
+    it('< 1 2 3 4 5 > should have 7 li', async () => {
+      await setup({ attributes: { current: 1, totalPages: 5 } });
+
+      const linkList = await page.findAll('osds-pagination >>> li');
+      expect(linkList.length).toBe(7);
+    });
+
+    it('< 1 2 3 4 5 6 > should have 8 li', async () => {
+      await setup({ attributes: { current: 1, totalPages: 6 } });
+
+      const linkList = await page.findAll('osds-pagination >>> li');
+      expect(linkList.length).toBe(8);
+    });
+
+    it('< 1 2 3 4 5 … 7 > should have 8 li', async () => {
+      await setup({ attributes: { current: 1, totalPages: 7 } });
+
+      const linkList = await page.findAll('osds-pagination >>> li');
+      expect(linkList.length).toBe(9);
+    });
+
+    it('< 1 2 3 4 5 … 8 > should have 8 li', async () => {
+      await setup({ attributes: { current: 1, totalPages: 8 } });
+
+      const linkList = await page.findAll('osds-pagination >>> li');
```

Comment:
in all theses kinf of tests, you want to test if the elements are correctly in DOM:
- the clickable buttons so, not just the li elements